### PR TITLE
define default prefix constants

### LINKED ISSUE #6506: Producing bottle block gives different results on macOS
**Please note that we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] ran a `brew` command and reproduced the problem with multiple formulae? If it's a problem with a single, official formula (not cask) please file this issue at Homebrew/homebrew-core: https://github.com/Homebrew/homebrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [ ] ran `brew config` and `brew doctor` and included their output with your issue?

# What you were trying to do (and why)

I run `brew bottle --merge` on macOS with two bottle JSONs, one for mojave and one for linux.
I mean `brew test-bot` run it and I found the issue in this command.

# What happened (include command output)

I don't have access to macOS machine (besides CI), so I replaced `Homebrew::DEFAULT_PREFIX` on linux to `/usr/local` just for testing and to prove this bug is real. But I also attach links to CI jobs where this bug occurs and where it is not appearing.

Here is where the bug occurs:
https://github.com/dawidd6/homebrew-tap/runs/240053060#step:5:222
Here it does not occur:
https://github.com/dawidd6/homebrew-tap/runs/240054363#step:5:223

See the order of arguments passed to `bottle --merge`. Seems like passing mojave first adds the additional unnecessary `prefix` line.

I was able to reproduce this locally:
<details>
  <summary>Command output</summary>
  <pre>
  ➜  dawidd6/brew master ✗  bin/brew bottle --merge ~/Downloads/keybase-cli--4.4.1.x86_64_linux.bottle.json ~/Downloads/keybase-cli--4.4.1.mojave.bottle.json
/usr/local
==> dawidd6/tap/keybase-cli
  bottle do
    root_url "https://dl.bintray.com/dawidd6/bottles-tap"
    cellar :any_skip_relocation
    rebuild 1
    sha256 "6f90ff414529a6683e0f954991d93280011b7d549ef1c426ecc5102a927fe6bd" => :mojave
    sha256 "4fc93e20f54f13399612b3d2eec0e0404658408cefd4b4c3a361908e6e8eabaf" => :x86_64_linux
  end
➜  dawidd6/brew master ✗  bin/brew bottle --merge ~/Downloads/keybase-cli--4.4.1.mojave.bottle.json ~/Downloads/keybase-cli--4.4.1.x86_64_linux.bottle.json
/usr/local
==> dawidd6/tap/keybase-cli
  bottle do
    root_url "https://dl.bintray.com/dawidd6/bottles-tap"
    prefix "/home/linuxbrew/.linuxbrew"
    cellar :any_skip_relocation
    rebuild 1
    sha256 "6f90ff414529a6683e0f954991d93280011b7d549ef1c426ecc5102a927fe6bd" => :mojave
    sha256 "4fc93e20f54f13399612b3d2eec0e0404658408cefd4b4c3a361908e6e8eabaf" => :x86_64_linux
  end
  </pre>
</details>

# What you expected to happen

A bottle block without `prefix` as it should be.

# Step-by-step reproduction instructions (by running `brew` commands)

See above output. The bug does not happen when running `brew` on linux, only on macOS.

### PR DESCRIPTION
**Pull request by dawidd6**
Closes #6506

### CODE REVIEW

File: Library/Homebrew/global.rb [hunk 1]
```diff
@@ -59,7 +59,10 @@
 module Homebrew
   extend FileUtils
 
-  DEFAULT_PREFIX ||= "/usr/local"
+  HOMEBREW_DEFAULT_PREFIX = "/home/linuxbrew/.linuxbrew"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/global.rb]: Wrong way around. Also, use this variable in `Library/Homebrew/os/linux/global.rb` too.

Comment (dawidd6) [reply Library/Homebrew/global.rb]: Oh snap, you're right, my bad.

I get this if I use those constants in `Library/Homebrew/os/linux/global.rb`:
```
/home/dawidd6/github/dawidd6/brew/Library/Homebrew/os/linux/global.rb:7:in `<module:Homebrew>': uninitialized constant Homebrew::LINUXBREW_DEFAULT_PREFIX (NameError)
	from /home/dawidd6/github/dawidd6/brew/Library/Homebrew/os/linux/global.rb:3:in `<top (required)>'
	from /home/dawidd6/github/dawidd6/brew/Library/Homebrew/os/global.rb:3:in `require'
	from /home/dawidd6/github/dawidd6/brew/Library/Homebrew/os/global.rb:3:in `<top (required)>'
	from /home/dawidd6/github/dawidd6/brew/Library/Homebrew/global.rb:57:in `require'
	from /home/dawidd6/github/dawidd6/brew/Library/Homebrew/global.rb:57:in `<top (required)>'
	from /home/dawidd6/github/dawidd6/brew/Library/Homebrew/brew.rb:23:in `require_relative'
	from /home/dawidd6/github/dawidd6/brew/Library/Homebrew/brew.rb:23:in `<main>'
```

Comment (MikeMcQuaid) [reply Library/Homebrew/global.rb]: You may need to update the namespace to ensure it's fine. It's definitely possible as it's included after.

File: Library/Homebrew/dev-cmd/bottle.rb [hunk 2]
```diff
@@ -16,7 +16,7 @@
     <% if !root_url.start_with?(HOMEBREW_BOTTLE_DEFAULT_DOMAIN) %>
     root_url "<%= root_url %>"
     <% end %>
-    <% if ![Homebrew::DEFAULT_PREFIX, "/usr/local"].include?(prefix) %>
+    <% if ![Homebrew::HOMEBREW_DEFAULT_PREFIX, Homebrew::LINUXBREW_DEFAULT_PREFIX].include?(prefix) %>
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/bottle.rb]: These are no longer declared in the `Homebrew` module

Comment (dawidd6) [reply Library/Homebrew/dev-cmd/bottle.rb]: Fixed.

**Comment by MikeMcQuaid** [comment]
Thanks again @dawidd6!

**Pull request merged**